node/pkg/node/node.go

```diff
@@ -0,0 +1,575 @@
+	return &GuardianOption{
+		name: "governor",
+		f: func(ctx context.Context, logger *zap.Logger, g *G) error {
+			if governorEnabled {
+				logger.Info("chain governor is enabled")
+				g.gov = governor.NewChainGovernor(logger, g.db, g.env)
+			} else {
+				logger.Info("chain governor is disabled")
+			}
+			return nil
+		}}
+}
+
+func GuardianOptionStatusServer(statusAddr string) *GuardianOption {
+	return &GuardianOption{
+		name: "status-server",
+		f: func(ctx context.Context, logger *zap.Logger, g *G) error {
+			if statusAddr != "" {
+				// Use a custom routing instead of using http.DefaultServeMux directly to avoid accidentally exposing packages
+				// that register themselves with it by default (like pprof).
+				router := mux.NewRouter()
+
+				// pprof server. NOT necessarily safe to expose publicly - only enable it in dev mode to avoid exposing it by
+				// accident. There's benefit to having pprof enabled on production nodes, but we would likely want to expose it
+				// via a dedicated port listening on localhost, or via the admin UNIX socket.
+				if g.env == common.UnsafeDevNet {
+					// Pass requests to http.DefaultServeMux, which pprof automatically registers with as an import side-effect.
+					router.PathPrefix("/debug/pprof/").Handler(http.DefaultServeMux)
+				}
+
+				// Simple endpoint exposing node readiness (safe to expose to untrusted clients)
+				router.HandleFunc("/readyz", readiness.Handler)
+
+				// Prometheus metrics (safe to expose to untrusted clients)
+				router.Handle("/metrics", promhttp.Handler())
+
+				go func() {
+					logger.Info("status server listening", zap.String("status_addr", statusAddr))
+					// SECURITY: If making changes, ensure that we always do `router := mux.NewRouter()` before this to avoid accidentally exposing pprof
+					logger.Error("status server crashed", zap.Error(http.ListenAndServe(statusAddr, router))) // #nosec G114 local status server not vulnerable to DoS attack
+				}()
+			}
+			return nil
+		}}
+}
+
+type IbcWatcherConfig struct {
+	Websocket string
+	Lcd       string
+	Contract  string
+}
+
+func GuardianOptionWatchers(watcherConfigs []watchers.WatcherConfig, ibcWatcherConfig *IbcWatcherConfig) *GuardianOption {
+	return &GuardianOption{
+		name: "watchers",
+		f: func(ctx context.Context, logger *zap.Logger, g *G) error {
+
+			chainObsvReqC := make(map[vaa.ChainID]chan *gossipv1.ObservationRequest)
+
+			chainMsgC := make(map[vaa.ChainID]chan *common.MessagePublication)
+
+			for _, chainId := range vaa.GetAllNetworkIDs() {
+				chainMsgC[chainId] = make(chan *common.MessagePublication)
+				go func(c <-chan *common.MessagePublication, chainId vaa.ChainID) {
+					zeroAddress := vaa.Address{}
+					for {
+						select {
+						case <-ctx.Done():
+							return
+						case msg := <-c:
+							if msg.EmitterChain != chainId {
+
+								logger.Error("SECURITY CRITICAL: Received observation from a chain that was not marked as originating from that chain",
```

Comment:
I added a test in https://github.com/wormhole-foundation/wormhole/blob/ed009d0976fe38f9a72abe8411c8746b6d735700/node/pkg/node/node_test.go#L334 that covers this code path. If it was logger.Fatal() here, then that would cause an os.Exit(), which we cannot catch in the test. We could dynamically overwrite logger.Fatal() in the test such that it doesn't call os.Exit() but I think logger.Error() is actually more appropriate here because the error is recoverable. 
The reason we did logger.Fatal() before was not because it was an irrecoverable error but because we wanted to nudge operators to take a look at and report this error. I think the correct thing to do would be to instead have operators monitor logs for "SECURITY CRITICAL".

Reply:
I disagree, but I made the change as requested.